### ISSUE TITLE
Support for generate command

### ISSUE THREAD

**Issue by polterguy** [dotnet/cli#10996]
Somebody sent me a pull request to my [own repo](https://github.com/polterguy/magic) supplying a CLI template file, and I started playing around with the CLI arguably for the first time really studying it myself, and I quickly realized the beauty of the dotnet CLI project, allowing me to among other things _"generate"_ entire solution/project structures, etc.

However, as I look through the project, I realize at least for my own purposes, how extremely useful it would be to have something resembling _"ng generate"_ in the dotnet CLI, which unfortunately doesn't seem to exist at the moment.

For instance for my personal usage in [Magic](https://github.com/polterguy/magic), being able to _"generate"_ new controllers/services/etc by using the dotnet CLI, would significantly reduce the pain associated with adding to the existing code base of a Magic project.

To create a new domain type in Magic for instance, requires to create a new service, a new service interface, a new controller endpoint, and the database model + the view model. Plus (possibly) also new projects for each of the previously mentioned classes/interfaces. If the CLI could simply _"automagically"_ create this for me, and my users, and either associate with existing project, and/or create new projects and associate the solution with - Obviously a lot of time would be saved in this process ...

Basically, what I am looking for, is something similar to `ng generate`, which allows the developer to generate components, etc - Hopefully implemented in such a way that I could extend it myself somehow, to work with my own Magic project ...

* Is this something that has been considered implement in the dotnet CLI?
* If not, why?
* Is it possible to create as my own plugin to the CLI?
* Would you possibly be willing to accept a pull request, if I actually went forth and implemented such commands myself, assuming it's generic in nature, good solid code, obeys by your coding standard, and not in any ways _"hard coded"_ towards my own baby (Magic) ...?

Thx,

.t

**Comment by polterguy** [comment]
In fact, as I thought about this for a while, I realize it can be done by creating the possibility of creating new verbs through some sort of plugin mechanism, allowing developers to extend dotnet CLI for themselves as they see fit ...

**Comment by livarcocc** [comment]
I would suggest you look into local tools. You could created your own local tool that could then be invoked through the dotnet CLI.

We currently have no plans to provide such a command. Our code generation capabilities are limited to dotnet new and dotnet add/remove commands that create simple projects/files or make small modifications to your sln or csproj.

**Comment by polterguy** [comment]
Thx, _"local tools"_? Link ...?

**Comment by livarcocc** [comment]
@wli3 @KathleenDollard do you have some useful links to share for local tools?

**Comment by wli3** [comment]
https://github.com/dotnet/cli/issues/10288 this is a good link today

**Comment by KathleenDollard** [comment]
@polterguy It sounds like you are looking at a template system. In .NET Core that is done with `dotnet new`. You can find out more about creating your own templates at https://docs.microsoft.com/en-us/dotnet/core/tutorials/create-custom-template.

**Comment by polterguy** [comment]
@KathleenDollard The `dotnet new` command is an awesome feature in the dotnet CLI project, but it only allows me to create _new_ projects. I need something that can _modify_ existing projects. To understand what I mean, Google _"ng generate"_ ...

Thx @all
